### PR TITLE
Indexer: Gazelle Games API

### PR DESCRIPTION
**Pull request by mikeoscar2006**
An API Port of Gazelle Games Tracker inspired by the existing HTML version and the Gazelle Base Tracker.
Solves Issue: #13544

### CODE REVIEW

**Comment by ilike2burnthing** [comment]
Why was genre support removed from GazelleTracker?

**Comment by mikeoscar2006** [comment]
On the base GazelleTracker, I've only made changes regarding support for AuthKey and different Header name for API Key.

On the GazelleGamesAPI Tracker file, all the categories are kept same as the HTML counterpart which is searchable through artistname parameter.

**Comment by garfield69** [comment]
@mikeoscar2006  did you remember to refresh your source before making changes?
your update appears to have dropped a prior  commit.

**Comment by mikeoscar2006** [comment]
> @mikeoscar2006 did you remember to refresh your source before making changes? your update appears to have dropped a prior commit.

Yes I did. For all the three pulls I made today.

**Comment by garfield69** [comment]
Very odd.
@ilike2burnthing  can you reapply your commit?

**Comment by mikeoscar2006** [comment]
I forgot to add 2 files in the commit, so I amended it. Maybe in that time gap a commit might've happened. I can re-open the pull with all the changes if it's better?

**Comment by garfield69** [comment]
I will add my commit from 10th august back into your pull. Somehow your source got out of date, I don't understand how.

**Comment by mikeoscar2006** [comment]
> I will add my commit from 10th august back into your pull. Somehow your source got out of date, I don't understand how.

Yeah, very strange indeed. Thanks for adding the commits. Once you test the indexers and approve this, I'll create a pull for the trackers that can benefit from this.

**Comment by garfield69** [comment]
@mikeoscar2006 
None of the Jackett team have an account on GGn and registration is closed ATM so I cannot test this.
Of the indexers that use the GazelleTracker abstract, I have access to Anthelion, Oppaitime, Orpheus, AlphaRatio and GreatPosterWall.
Oppaitime (uid/pwd) is down so I can not test ATM.
Anthelion (uid/pwd/2fa), AlphaRatio (uid/pwd), and GPW (uid/pwd) are working fine.
Orpheus (apikey) only works in production and fails with this PR with an invalid apikey error.
see the maintainers page for the key so you can test/debug

**Comment by mikeoscar2006** [comment]
Hey @garfield69 
Thanks for the key for debugging, but it works for me without any problems. Steps Followed:
1. Configure Indexer with API Key
2. Make a Blank Query Search
3. Download a torrent file from the first result.

It works for me without an issue in any of these steps. Make sure you have all three files from the pull and Let me know if it still creates a problem for you.

File: src/Jackett.Common/Indexers/Abstract/GazelleTracker.cs
Comment (garfield69) [inline src/Jackett.Common/Indexers/Abstract/GazelleTracker.cs:233]: OK. did some debugging and discovered that orpheus was failing with `your api key did not work` 
but not because the apikey was actually invalid, or not picked up from the config correctly,
but because the subsequent test to perform a keyworldless query as part of the config validation was failing to process the returned json of the search results.
my log shows that the error was at line 233 while processing the grouptime.

After I restored the previous lines 226-230 to replace 233 my indexer started working again.

Please restore that grouptime logic

**Review by garfield69 — CHANGES_REQUESTED** [review]

File: src/Jackett.Common/Indexers/Abstract/GazelleTracker.cs
Comment (mikeoscar2006) [reply src/Jackett.Common/Indexers/Abstract/GazelleTracker.cs]: Restored. Probably some leftover code from my migration. It might've not been affecting all results that are why in my testing of Orpheus, It worked for me.

Nevertheless, Made the changes and Tested them on Orpheus and GGN, With and Without Query + Downloads.

**Review by garfield69 — APPROVED** [review]

**Comment by garfield69** [comment]
@ItsME6969  since none of the team are members we really could use someone to test the GazelleGamesApi
please tell me you are not a docker user and that you would be willing to test this new indexer?

**Comment by ItsME6969** [comment]
@garfield69 unfortunately I'm using docker and away from home for the next month although i can access my jackett remotely but I can't access the actual site from where I'm at right now since they don't like folks using vpn so I'm not sure even how I could help.

**Comment by garfield69** [comment]
@mikeoscar2006  @ilike2burnthing  vote: do we push it out (since its a new indexer) and see if any one using it reports a problem or do we put it on hold until ItsME6969 is available again?

**Comment by ItsME6969** [comment]
Don't wait on me 😂 I would just push it out and other people with accounts who encounter errors can report back and we can also use them for testing. I can run basic searches remotely if that will help with regards to testing.

**Comment by mikeoscar2006** [comment]
I say we push out and wait for reports since I already have access, I can probably resolve the issues later.

**Comment by ilike2burnthing** [comment]
Sorry for the delay, I was dead to the world. Yes, I say push it out, and as ItsME6969 can access their Jackett, they can report back if there are any issues.

**Pull request merged**
